### ISSUE TITLE
Gattling Tank EXP Requirement Balancing

### ISSUE THREAD

**Issue by ExiLe-ZH** [TheSuperHackers/GeneralsGamePatch#1175]
Unit | Regular | Veteran | Elite | Hero
--  | -- | -- | -- | --
Gatling Tank | 0 | 80 | 200 | 300
Quad Cannon | 0 | 100 | 150 | 300
Humvee | 0 | 100 | 150 | 300
Toxin Tractor | 0 | 100 | 150 | 300
Dragon Tank | 0 | 80 | 200 | 300

Now the balance between Gats and Quads should massively improve with the Quad Damage Distribution change
* #1055

To further improve this the Experience Requirements can't be overlooked. When we compare the requirements for Gats and Quads the balance doesn't add up very logically. Here a few facts:
- Vanilla Gats have a 20xp advantage in reaching Veteran status.
- Veteran Quad have a 50xp advantage in reaching Elite status.
- Quads can kill buildings to gain experience really fast.
- Quads can collect scraps to upgrade it's gun.

With these facts in mind we can conclude that Quads have a much better potential at getting strong fast. The only advantage Vanilla Gats have is the 20xp reduction for Veteran level.

Few more things to note:
- The Gat Veteran requirement of 80 is an inconsistency, for every other unit except Dragons it's 100xp.
- Gat Veteran nerf to 100 would be a buff for Tank, a nerf for VChina/Nuke.
- Gats need advantages across the board to compete?

### Proposal 1
First proposal would be to give Gats some advantage at each level:
`0 80 140 280`

### Proposal 2
To play it safer bring down Gat Elite Requirement to 150 (from 200):
`0 80 150 300`

### Proposal 3
Quad Elite Requirement to 200:
`0 100 200 300`

Gat Elite Requirement to 180 (and Hero to 280):
`0 80 180 300` or `0 80 180 280`

**Comment by Jundiyy** [comment]
Would prefer 2 over 1, although don't know if I prefer anything at all at the moment. Been reading the recent topics and we are changing quite a few of these things, need to see what adds up.

**Comment by ExiLe-ZH** [comment]
Which ones make China OP?

**Comment by Stubbjax** [comment]
Although the Gattling Tank's transition from veteran to elite is rather slow / awkward, such a change would mostly benefit China Tank over vChina and Nuke - which is really the opposite of what is needed.

**Comment by ExiLe-ZH** [comment]
Tank could also use it, it has a fairly low winrate overall.

120xp is quite alot to become Elite, 3 Vee/Quad/Chinook kills, kinda unrealistic amount.

Alternative is making Quad Elite 200 and Gat Elite 180, I'm less of a fan of that, but it still would be a balance improvement.

**Comment by commy2** [comment]
CCG curve for comparision:

Object ChinaTankGattling
  ExperienceRequired = 0 100 150 300

The reason the Gatling tank curve is so weird is probably EA attempting to balance it in the transition from CCG to ZH. This change was introduce alongside reduing the damage by 25% (20 -> 15) and range (170 -> 150), as well as reducing the spin up time from 3 to 2 shots.

EA basically has no idea and fumbled around until we are were we are.

**Comment by Stubbjax** [comment]
> Tank could also use it, it has a fairly low winrate overall.

It seems to average ~50% across all match ups at both general and high skill levels based on the 2022 1v1 stats.

![image](https://user-images.githubusercontent.com/11547761/189475899-f071c203-ecd3-4743-8b57-259ab7b13f10.png)

![image](https://user-images.githubusercontent.com/11547761/189475909-b1e2520a-fc37-41d8-aac8-f3f7d92e8415.png)

**Comment by ExiLe-ZH** [comment]
> CCG curve for comparision:
> 
> Object ChinaTankGattling ExperienceRequired = 0 100 150 300
> 
> The reason the Gatling tank curve is so weird is probably EA attempting to balance it in the transition from CCG to ZH. This change was introduce alongside reduing the damage by 25% (20 -> 15) and range (170 -> 150), as well as reducing the spin up time from 3 to 2 shots.
> 
> EA basically has no idea and fumbled around until we are were we are.

I don't think 0 100 150 300 would be a good idea, this is maybe too much of a buff for Tank and a nerf for VChina/Nuke.

**Comment by ExiLe-ZH** [comment]
> It seems to average ~50% across all match ups at both general and high skill levels based on the 2022 1v1 stats.
> 
> ![image](https://user-images.githubusercontent.com/11547761/189475899-f071c203-ecd3-4743-8b57-259ab7b13f10.png)
> 
> ![image](https://user-images.githubusercontent.com/11547761/189475909-b1e2520a-fc37-41d8-aac8-f3f7d92e8415.png)

On average level Tank indeed performs well because it's easy to play.

However when we compare that to the Zero Hour Russian Ladder, which is Semi level and above (1550+ ELO) we see this:

```
1.  Air -----  60% G:402 W:241 L:161 
2.  Inf -----  60% G:435 W:259 L:176 
3.  Tox -----  56% G:462 W:261 L:201 
4.  Stealth -  56% G:505 W:284 L:221 
5.  GLA -----  52% G:532 W:274 L:258 
6.  USA -----  52% G:494 W:258 L:236 
7.  Laser ---  51% G:408 W:207 L:201 
8.  Demo ----  49% G:443 W:216 L:227 
9.  Tank ----  46% G:450 W:205 L:245 
10. SWG -----  42% G:434 W:182 L:252 
11. Nuke ----  40% G:377 W:152 L:225 
12. China ---  36% G:478 W:171 L:307 
```

**Comment by Stubbjax** [comment]
Those numbers support my initial comment. Ideally the primary benefactors are vChina and Nuke, and/or Tank secondary.

**Comment by ExiLe-ZH** [comment]
I agree that most focus should be on VChina/Nuke in general, however the Veteran Requirement is already in a pretty good spot. While an Elite buff does affect Tank more directly ofcourse, I think a (small) buff wouldn't hurt, proposal 3 for example. 

Proposal 3 would make most sense if Vee Elite Requirement is also put at 200, then we have a nice new consistency where all these units are based around the 0 100 200 300 model. Only the Tox Tractor would be able to vet up a little faster at 150xp, but this is justified as it has far less killing potential than the others (Hero could also be put at 250 probably).

**Comment by RisingZH** [comment]
You have to be careful with the rekja expert stats as the sample sizes from each matchup varies massively. 

The thing with gats is you never use them to kill buildings so they miss out on a lot of exp that quads get, which is why you always see vetted quads and rarely vetted gats.

On a slight tangent, has changing how 'spinning up gats' works been discussed anywhere (eg force firing the ground before attacking)? On ccg it can be done very effectively (probably too much) but on ZH the spin doesn't last long enough to really do anything. I think it could be a pretty cool change; increases micro potential (always good) and makes gats better defensively (to counter no eco's/tech rpgs).

**Comment by ExiLe-ZH** [comment]
If you press Stop while your Gats are force firing and they instantly retarget a unit in range, is that already enough delay for the gun having to spin up again? 

Does manually retarget a unit in range have less delay? 

It's an interesting discussion anyway, let's open one.

**Comment by commy2** [comment]
> On a slight tangent, has changing how 'spinning up gats' works been discussed anywhere (eg force firing the ground before attacking)? On ccg it can be done very effectively (probably too much) but on ZH the spin doesn't last long enough to really do anything. I think it could be a pretty cool change; increases micro potential (always good) and makes gats better defensively (to counter no eco's/tech rpgs).

This so much. The unit is so much more involved in CCG thanks to this.

> Does manually retarget a unit in range have less delay?

In CCG, the Gat loses spin up if it doesn't shoot for 2 seconds. In ZH, it already does this after 1 second. On the other hand, in CCG the Gat spins up to the next stage after 3 and then 6 shots total. In ZH, it is 2 and then 6 shots total.

There are also various issues with how veterancy and spin up interact. Some combinations may increase DPS more than expected, and some stages don't change DPS at all.

**Comment by Stubbjax** [comment]
> On a slight tangent, has changing how 'spinning up gats' works been discussed anywhere (eg force firing the ground before attacking)? On ccg it can be done very effectively (probably too much) but on ZH the spin doesn't last long enough to really do anything. I think it could be a pretty cool change; increases micro potential (always good) and makes gats better defensively (to counter no eco's/tech rpgs).

Definitely agree with the barrel spin suggestion. https://github.com/TheSuperHackers/GeneralsGamePatch/issues/640#issuecomment-1017311040

**Comment by ExiLe-ZH** [comment]
Should we extent this topic to Vees, Quads and Gats?

The 0 100 150 300 model doesn't make too much sense does it? First having to gain 100, then 50 and then 150?

I don't like the idea of nerfing to much, but don't forget that Advanced Training makes things even more snowbally for TowVee spam.

Maybe we can come up with a model that fits all without buffing/nerfing anything too hard.

**Comment by Jundiyy** [comment]
I wouldn't touch the Quads and Vees (maybe a tiny bit), they are main units and a small change could make a massive change, better to work on the Gat Tank.

Regarding benefitting Tank over China and Nuke, I don't think it would be a massive problem if you mean Tank vs China/Nuke, Nuke does better anyway and China can be map dependant, if it can hold off 1 or 2 sides then it can sit and bunker.

Of course, Tank vs USAs could change and give more benefit to Tank.

**Comment by ExiLe-ZH** [comment]
What I mean by benefiting Tank over China/Nuke is that Tank "feels" a Vet 2 Requirement change more directly.

**Comment by xezon** [comment]
We can have different XP steps for regular China Gatling and Tank Gatling. We did the same with Regular Lotus and Super Lotus.

**Comment by ExiLe-ZH** [comment]
could give us the opportunity to finetune things better.
